File: src/main/java/pl/patrykdolata/lifemanager/service/UserService.java
package pl.patrykdolata.lifemanager.service;

import pl.patrykdolata.lifemanager.model.JwtToken;
import pl.patrykdolata.lifemanager.model.LoginInfo;
import pl.patrykdolata.lifemanager.model.NewUser;

public interface UserService {
    JwtToken authorize(LoginInfo loginInfo);

    void register(NewUser user);
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/service/UserService.kt
package pl.patrykdolata.lifemanager.service

import pl.patrykdolata.lifemanager.model.LoginInfo
import pl.patrykdolata.lifemanager.model.NewUser
import pl.patrykdolata.lifemanager.model.User

interface UserService {
    fun authorize(loginInfo: LoginInfo): User

    fun register(user: NewUser)
}
